src/rgw/rgw_rados.cc

```diff
@@ -145,6 +145,33 @@ static bool rgw_obj_to_raw(const RGWZoneGroup& zonegroup, const RGWZoneParams& z
   return rgw_get_obj_data_pool(zonegroup, zone_params, placement_id, obj, &raw_obj->pool);
 }
 
+bool rgw_s3_is_multipart_encrypted(
+  std::map<std::string, ceph::bufferlist>& attrs)
+{
+  auto attr_iter1 = attrs.find(RGW_ATTR_CRYPT_MODE);
+  if (attr_iter1 == attrs.end()) {
+    return false;
+  }
+  auto attr_iter2 = attrs.find(RGW_ATTR_MANIFEST);
+  if (attr_iter2 == attrs.end() || attr_iter2->second.length() == 0) {
+    return false;
+  }
+  RGWObjManifest manifest;
+  bufferlist::iterator miter = attr_iter2->second.begin();
+  decode(manifest, miter);
```

Comment:
how about:
```
  RGWObjManifest manifest;
  try {
    bufferlist::iterator miter = attr_iter2->second.begin();
    decode(manifest, miter);
  } catch (buffer::error& err) {
    ldout(cct, 0) << "ERROR: couldn't decode manifest" << dendl;
    return false;
  }
```

Reply:
iirc, some buffer operators are among the only interfaces that throw c++ exceptions, so this is probably sensible